fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/builder/SoundTypeBuilder.java

```diff
@@ -0,0 +1,183 @@
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.builder;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.include.com.google.common.base.Preconditions;
+
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.StringIdentifiable;
+
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricSoundsProvider;
+import net.fabricmc.fabric.impl.datagen.SoundTypeBuilderImpl;
+
+/**
+ * Provides functionality for building entries that go into a sounds.json file.
+ *
+ * <p>Use in conjunction with {@link FabricSoundsProvider} to generate sound definitions.
+ *
+ * @see net.minecraft.client.sound.SoundManager
+ * @see net.minecraft.client.sound.WeightedSoundSet
+ */
+public interface SoundTypeBuilder {
+	/**
+	 * Creates a new builder pre-filled with a subtitle translation string based on the passed event.
+	 *
+	 * @param event The sound event.
+	 * @return New sound type builder
+	 */
+	static SoundTypeBuilder of(SoundEvent event) {
+		Preconditions.checkArgument(event != null, "Sound event must not be null.");
+		return of().subtitle("subtitles." + event.id().getNamespace() + "." + event.id().getPath());
+	}
+
+	/**
+	 * Creates a new empty builder.
+	 *
+	 * @param event The sound event.
+	 * @return New sound type builder
+	 */
+	static SoundTypeBuilder of() {
+		return new SoundTypeBuilderImpl();
+	}
+
+	/**
+	 * Sets the sound category the sound event must play on.
+	 *
+	 * <p>The default category is {@link SoundCategory#NEUTRAL}. GUI elements should use {@link SoundCategory#MASTER}.
+	 */
+	SoundTypeBuilder category(SoundCategory category);
+
+	/**
+	 * Sets an optional translation string to use for the sound's subtitle.
+	 *
+	 * <p>The default is null (no subtitle).
+	 */
+	SoundTypeBuilder subtitle(@Nullable String subtitle);
+
+	/**
+	 * Adds one sound to the event.
+	 */
+	SoundTypeBuilder sound(EntryBuilder sound);
+
+	/**
+	 * Adds one or more sounds to the event.
+	 * This is a shorthand method for quickly adding multiple entries where
+	 * each sound is a variant with an index.
+	 *
+	 * @param sound The base sound to add.
+	 * @param count The number of instances of that sound to register.
+	 */
+	SoundTypeBuilder sound(EntryBuilder sound, int count);
+
+	/**
+	 * @see net.minecraft.client.sound.Sound.RegistrationType
+	 */
+	enum RegistrationType implements StringIdentifiable {
+		FILE("file"),
+		SOUND_EVENT("event");
+
+		public static final Codec<RegistrationType> CODEC = StringIdentifiable.createCodec(RegistrationType::values);
+
+		private final String name;
+
+		RegistrationType(String name) {
+			this.name = name;
+		}
+
+		@Override
+		public String asString() {
+			return name;
+		}
+	}
+
+	interface EntryBuilder {
+		/**
+		 * Creates a builder for constructing a new sound entry.
+		 *
+		 * @param name The Identifier of the sound file or event this entry must reference.
```

Comment:
"identifier" should not be capitalized; however this should be rewritten as this is not clear. Also note general `@param` note (all lowercase, no period at end)